Segmentation fault on Hyperion 2.0.0-alpha.7

<!-- Please don't delete this template or we'll close your issue -->
<!-- Before creating an issue please make sure you are using the latest version of Hyperion. -->

<!-- Please confirm you will submit an issue. -->
<!-- Issues which contain questions or support requests will be closed. -->
<!-- (Update "[ ]" to "[x]" to check a box) -->

- [x] I confirm that this is an issue rather than a question.

<!-- Please ask questions here -->
<!-- https://hyperion-project.org -->

## Bug report
Hyperion 2.0.0-alpha.7 crashes after setting up LED hardware with a segmentation fault

#### Steps to reproduce

1. Flash [HyperBian](https://github.com/hyperion-project/HyperBian/releases/tag/HyperBian)
2. Open Hyperion UI from the browser
3. Setup LED Hardware:
- LED Controller
  - Controller Type: `adalight`
  - Output path: `/dev/ttyUSB0`
4. Save Settings

#### What is expected?
Hyperion running

#### What is actually happening?
Hyperion crashes as soon as the Settings are saved. Because by default Hyperion runs as service, Hyperion restarts and crashes again, and again ...

Stopping the service and manual starting `hyperiond` shows these logs:

```
2020-07-25T11:13:20.076 hyperiond EFFECTENGINE : <ERROR> ###### PYTHON EXCEPTION ######
2020-07-25T11:13:20.077 hyperiond EFFECTENGINE : <ERROR> ## In effect 'Rainbow swirl fast'
2020-07-25T11:13:20.077 hyperiond EFFECTENGINE : <ERROR> ## ZipImportError: can't decompress data; zlib not available
2020-07-25T11:13:20.843 hyperiond EFFECTENGINE : <ERROR> ## Traceback (most recent call last):
2020-07-25T11:13:20.844 hyperiond EFFECTENGINE : <ERROR> ## File "<string>", line 1, in <module>
2020-07-25T11:13:20.845 hyperiond EFFECTENGINE : <ERROR> ## File "/usr/share/hyperion/bin/python35.zip/random.py", line 40, in <module>
    from types import MethodType as _MethodType, BuiltinMethodType as _BuiltinMethodType
2020-07-25T11:13:20.846 hyperiond EFFECTENGINE : <ERROR> ## File "/usr/share/hyperion/bin/python35.zip/types.py", line 166, in <module>
    import functools as _functools
2020-07-25T11:13:20.846 hyperiond EFFECTENGINE : <ERROR> ## File "/usr/share/hyperion/bin/python35.zip/functools.py", line 21, in <module>
    from collections import namedtuple
2020-07-25T11:13:20.847 hyperiond EFFECTENGINE : <ERROR> ## File "/usr/share/hyperion/bin/python35.zip/collections/__init__.py", line 32, in <module>
    from reprlib import recursive_repr as _recursive_repr
2020-07-25T11:13:20.848 hyperiond EFFECTENGINE : <ERROR> ## zipimport.ZipImportError: can't decompress data; zlib not available
2020-07-25T11:13:20.848 hyperiond EFFECTENGINE : <ERROR> ###### EXCEPTION END ######

Hyperion caught signal :SIGSEGV
2020-07-25T11:13:20.923 hyperiond CORE         : <ERROR> 	/lib/arm-linux-gnueabihf/libc.so.6(__default_rt_sa_restorer+0) [0x7561e130]
2020-07-25T11:13:20.924 hyperiond CORE         : <ERROR> 	/usr/share/hyperion/bin/../lib/libQt5Core.so.5(QRingBuffer::reserve(long long)+0x44) [0x75af1b74]
Segmentation fault
```

From above logs multiple errors occur, not sure if they are related. The `zipimport.ZipImportError: can't decompress data; zlib not available` error was reported earlier at #881 

Removing hyperion.db make Hyperion start again.

#### System
<!-- In the web interface of the Hyperion config go to System > About Hyperion and Paste the content of "System info (Github Issue)" here -->

```
Hyperion Server: 
- Build:       (HEAD detached at ffa2fd45) (Paulchen-Panther-69e9fa6d/ffa2fd45-1595517126)
- Build time:  Jul 24 2020 15:49:41
- Git Remote:  https://github.com/hyperion-project/hyperion.ng
- Version:     2.0.0-alpha.7
- UI Lang:     auto (BrowserLang: en-gb)
- UI Access:   expert
- Avail Capt:  dispmanx,v4l2,framebuffer,qt

Hyperion Server OS: 
- Distribution: HyperBian GNU/Linux 10 (buster)
- Arch:         arm
- Kernel:       linux (5.4.51-v7+ (WS: 32))
- Browser:      Mozilla/5.0 (Macintosh; Intel Mac OS X 10_15_6) AppleWebKit/605.1.15 (KHTML, like Gecko) Version/13.1.2 Safari/605.1.15 
```

> @RaymondMouthaan I can reproduce the issue with alpha7 on HyperBian but building from master seems to fix the issue. Can you try the following steps ?
> 
> * Stop hyperion service, remove hyperion package from system
> * Build master from source, don't install, just execute built binary from build directory (`bin/hyperiond`). Does it still crash ?
> * If it doesn't crash, install it with `sudo make install`. Now run `/usr/local/bin/hyperion`, does it work OK ?
> * If it still crashes, configure the build with `cmake -DCMAKE_BUILD_TYPE=Debug`, build it with `make -j $(nproc)`.
> * Start the built binary from bin directory (`bin/hyperiond`). Does it work OK ?
> * If it doesn't work OK, start gdb and follow the steps mentioned by Paulchen above.
> * If it works, let me know please.
> 
> Thanks.

@m-seker,

Following the instructions in [CompileHowto.md](https://github.com/hyperion-project/hyperion.ng/blob/master/CompileHowto.md#debianubuntuwin10linuxsubsystem) I face the following error while trying to compile:

```
$ cmake -DCMAKE_BUILD_TYPE=Release ..
```

```
-- CMake Version: 3.13.4
CMake Deprecation Warning at CMakeLists.txt:18 (CMAKE_POLICY):
  The OLD behavior for policy CMP0026 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- PLATFORM is not defined, evaluated platform: rpi
-- PLATFORM: rpi
-- ENABLE_AMLOGIC = OFF
-- ENABLE_DISPMANX = ON
-- ENABLE_FB = ON
-- ENABLE_OSX = OFF
-- ENABLE_SPIDEV   = ON
-- ENABLE_TINKERFORGE = ON
-- ENABLE_V4L2 = ON
-- ENABLE_WS281XPWM = ON
-- ENABLE_AVAHI = ON
-- ENABLE_USB_HID = ON
-- ENABLE_CEC = ON
-- ENABLE_X11 = OFF
-- ENABLE_XCB = OFF
-- ENABLE_QT = ON
-- ENABLE_TESTS = OFF
-- ENABLE_PROFILER = OFF
-- ENABLE_EXPERIMENTAL = OFF
-- validate json files
   checked files: 81 success: 81 errors: 0
-- validate json effect files
   checked effect files: 39 success: 39 errors: 0
-- validate json file
-- Current Version: 2.0.0-alpha.7
--  - Build: master (GitHub-bdf00206/a18ccb8b-1596885757)
-- Linker: GNU gold
-- Found Qt Version: 5.11.3
-- Found libusb-1.0:
--  - Includes: /usr/include/libusb-1.0
--  - Libraries: /usr/lib/arm-linux-gnueabihf/libusb-1.0.so
-- Using Turbo JPEG library: /usr/lib/arm-linux-gnueabihf/libturbojpeg.so
CMake Error at dependencies/CMakeLists.txt:31 (add_subdirectory):
  The source directory

    /home/pi/hyperion-ng/hyperion.ng/dependencies/external/flatbuffers

  does not contain a CMakeLists.txt file.


CMake Error at dependencies/CMakeLists.txt:39 (export):
  export given target "flatc" which is not built by this project.


CMake Error at dependencies/CMakeLists.txt:48 (get_property):
  get_property could not find TARGET flatc.  Perhaps it has not yet been
  created.


-- Using flatbuffers compiler:
CMake Error at dependencies/CMakeLists.txt:99 (add_subdirectory):
  The source directory

    /home/pi/hyperion-ng/hyperion.ng/dependencies/external/protobuf

  does not contain a CMakeLists.txt file.


CMake Error at dependencies/CMakeLists.txt:107 (export):
  export given target "protoc_compiler" which is not built by this project.


CMake Error at dependencies/CMakeLists.txt:115 (get_property):
  get_property could not find TARGET protoc_compiler.  Perhaps it has not yet
  been created.


-- Using protobuf compiler:
-- Using static mbedtls libraries (build version "2.23.0")
-- CPACK: Found DEB builder
-- Configuring incomplete, errors occurred!
See also "/home/pi/hyperion-ng/hyperion.ng/CMakeFiles/CMakeOutput.log".
See also "/home/pi/hyperion-ng/hyperion.ng/CMakeFiles/CMakeError.log".
```

not sure what went wrong, but indeed the `hyperiond` is located in `hyperion.ng/bin` and not in `build`

Just to summaries
- `mkdir build`
- `cd build`
- `cmake -DCMAKE_BUILD_TYPE=Debug ..`

```
cmake -DCMAKE_BUILD_TYPE=Debug ..
-- CMake Version: 3.13.4
CMake Deprecation Warning at CMakeLists.txt:18 (CMAKE_POLICY):
  The OLD behavior for policy CMP0026 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- PLATFORM is not defined, evaluated platform: rpi
-- PLATFORM: rpi
-- ENABLE_AMLOGIC = OFF
-- ENABLE_DISPMANX = ON
-- ENABLE_FB = ON
-- ENABLE_OSX = OFF
-- ENABLE_SPIDEV   = ON
-- ENABLE_TINKERFORGE = ON
-- ENABLE_V4L2 = ON
-- ENABLE_WS281XPWM = ON
-- ENABLE_AVAHI = ON
-- ENABLE_USB_HID = ON
-- ENABLE_CEC = ON
-- ENABLE_X11 = OFF
-- ENABLE_XCB = OFF
-- ENABLE_QT = ON
-- ENABLE_TESTS = OFF
-- ENABLE_PROFILER = OFF
-- ENABLE_EXPERIMENTAL = OFF
-- validate json files
   checked files: 81 success: 81 errors: 0
-- validate json effect files
   checked effect files: 39 success: 39 errors: 0
-- validate json file
-- Current Version: 2.0.0-alpha.7
--  - Build: master (GitHub-bdf00206/a18ccb8b-1596885757)
-- Linker: GNU gold
-- Found Qt Version: 5.11.3
-- Found libusb-1.0:
--  - Includes: /usr/include/libusb-1.0
--  - Libraries: /usr/lib/arm-linux-gnueabihf/libusb-1.0.so
-- Using Turbo JPEG library: /usr/lib/arm-linux-gnueabihf/libturbojpeg.so
Proceeding with version: 1.12.0.103
-- Using flatbuffers compiler: /home/pi/hyperion-ng/hyperion.ng/bin/flatc
CMake Deprecation Warning at dependencies/external/protobuf/src/CMakeLists.txt:329 (cmake_policy):
  The OLD behavior for policy CMP0026 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- Using protobuf compiler: /home/pi/hyperion-ng/hyperion.ng/bin/protoc
-- Using static mbedtls libraries (build version "2.23.0")
-- Collecting Dependencies for target file: /home/pi/hyperion-ng/hyperion.ng/bin/hyperiond
-- CPACK: Found DEB builder
-- Configuring done
-- Generating done
-- Build files have been written to: /home/pi/hyperion-ng/hyperion.ng
```

- `make -j $(nproc)`
   - results in -> `make: *** No targets specified and no makefile found.  Stop.`
- build dir remains empty 

- cd to `hyperion.ng/`
- `make -j $(nproc)` starts compiling but the output goes to `/hyperion.ng/bin`

> Can you help further to diagnose the issue ? Please checkout alpha7 tag in the repository and try building/running again. If it runs correctly, then it means we didn't fix it :) There may be a dependency issue.

Happy do help any further!

Just to be sure ...

I removed `hyperion.ng` folder to start from scratch, then used following commands to checkout `alpha.7`

```
$ git clone https://github.com/hyperion-project/hyperion.ng.git
$ cd hyperion.ng
$ git checkout tags/2.0.0-alpha.7
$ mkdir build
$ cd build
$ cmake -DCMAKE_BUILD_TYPE=Release ..
```

This causes the following error:

```
cmake -DCMAKE_BUILD_TYPE=Release ..
-- CMake Version: 3.13.4
-- The C compiler identification is GNU 8.3.0
-- The CXX compiler identification is GNU 8.3.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Deprecation Warning at CMakeLists.txt:18 (CMAKE_POLICY):
  The OLD behavior for policy CMP0026 will be removed from a future version
  of CMake.

  The cmake-policies(7) manual explains that the OLD behaviors of all
  policies are deprecated and that a policy should be set to OLD only under
  specific short-term circumstances.  Projects should be ported to the NEW
  behavior and not rely on setting a policy to OLD.


-- Found Python3: /usr/bin/python3.7 (found version "3.7.3") found components:  Interpreter Development
-- PLATFORM is not defined, evaluated platform: rpi
-- PLATFORM: rpi
-- ENABLE_AMLOGIC = OFF
-- ENABLE_DISPMANX = ON
-- ENABLE_FB = ON
-- ENABLE_OSX = OFF
-- ENABLE_SPIDEV   = ON
-- ENABLE_TINKERFORGE = ON
-- ENABLE_V4L2 = ON
-- ENABLE_WS281XPWM = ON
-- ENABLE_AVAHI = ON
-- ENABLE_USB_HID = ON
-- ENABLE_CEC = ON
-- ENABLE_X11 = OFF
-- ENABLE_QT = ON
-- ENABLE_TESTS = OFF
-- ENABLE_PROFILER = OFF
-- validate json file
-- Current Version: 2.0.0-alpha.7
--  - Build: (HEAD detached at 2.0.0-alpha.7) (Paulchen-Panther-69e9fa6d/ffa2fd45-1595517126)
-- Performing Test COMPILER_SUPPORTS_CXX11
-- Performing Test COMPILER_SUPPORTS_CXX11 - Success
-- Performing Test COMPILER_SUPPORTS_CXX0X
-- Performing Test COMPILER_SUPPORTS_CXX0X - Success
-- Linker: GNU gold
-- Found Qt Version: 5.11.3
-- Found libusb-1.0:
--  - Includes: /usr/include/libusb-1.0
--  - Libraries: /usr/lib/arm-linux-gnueabihf/libusb-1.0.so
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Performing Test TURBOJPEG_WORKS
-- Performing Test TURBOJPEG_WORKS - Success
-- Found TurboJpeg: /usr/lib/arm-linux-gnueabihf/libturbojpeg.so
-- Using Turbo JPEG library: /usr/lib/arm-linux-gnueabihf/libturbojpeg.so
CMake Error at dependencies/CMakeLists.txt:31 (add_subdirectory):
  The source directory

    /home/pi/hyperion.ng/dependencies/external/flatbuffers

  does not contain a CMakeLists.txt file.


CMake Error at dependencies/CMakeLists.txt:39 (export):
  export given target "flatc" which is not built by this project.


CMake Error at dependencies/CMakeLists.txt:48 (get_property):
  get_property could not find TARGET flatc.  Perhaps it has not yet been
  created.


-- Using flatbuffers compiler:
CMake Error at dependencies/CMakeLists.txt:99 (add_subdirectory):
  The source directory

    /home/pi/hyperion.ng/dependencies/external/protobuf

  does not contain a CMakeLists.txt file.


CMake Error at dependencies/CMakeLists.txt:107 (export):
  export given target "protoc_compiler" which is not built by this project.


CMake Error at dependencies/CMakeLists.txt:115 (get_property):
  get_property could not find TARGET protoc_compiler.  Perhaps it has not yet
  been created.


-- Using protobuf compiler:
-- Found PythonInterp: /usr/bin/python3 (found suitable version "3.7.3", minimum required is "3")
-- Using static mbedtls libraries (build version "2.23.0")
-- Found CEC: /usr/include
-- CPACK: Found DEB builder
-- Configuring incomplete, errors occurred!
See also "/home/pi/hyperion.ng/build/CMakeFiles/CMakeOutput.log".
See also "/home/pi/hyperion.ng/build/CMakeFiles/CMakeError.log".
```

I guess `git submodule update --init --force` must be executed again ?

--- a/libsrc/leddevice/dev_serial/ProviderRs232.cpp
+++ b/libsrc/leddevice/dev_serial/ProviderRs232.cpp
@@ -138,7 +138,7 @@ bool ProviderRs232::tryOpen(int delayAfterConnect_ms)
 
 		_rs232Port.setBaudRate( _baudRate_Hz );
 
-		Debug(_log, "_rs232Port.open(QIODevice::WriteOnly): %s, Baud rate [%d]bps", QSTRING_CSTR(_deviceName), _baudRate_Hz);
+		Debug(_log, "_rs232Port.open(QIODevice::ReadWrite): %s, Baud rate [%d]bps", QSTRING_CSTR(_deviceName), _baudRate_Hz);
 
 		QSerialPortInfo serialPortInfo(_deviceName);
 
@@ -153,7 +153,7 @@ bool ProviderRs232::tryOpen(int delayAfterConnect_ms)
 
 		if (!serialPortInfo.isNull() )
 		{
-			if ( !_rs232Port.open(QIODevice::WriteOnly) )
+			if ( !_rs232Port.open(QIODevice::ReadWrite) )
 			{
 				this->setInError(_rs232Port.errorString());
 				return false;

